src/game.js

```diff
@@ -0,0 +1,67 @@
+'use strict';
+
+class Game {
+  constructor() {
+    this.frame = null;
+    this.turn = 0;
+    this.scorecard = [];
+    this.MAXIMUM_FRAME = 10;
+  }
+
+  beginFrame() {
+    if(this.turn >= this.MAXIMUM_FRAME) {
+      return this.finishGame();
+    } 
+    this.turn += 1;
+    this.frame = new Frame(this.turn);
+    return `Frame ${this.turn}`;
+  }
+
+  throwBall(points) {
+    return this.frame.addRoll(points);
+  }
+
+  addToScorecard() {
+    this.scorecard.push(this.frame.score);
+    return this.scorecard;
+  }
+
+  strikeBonus() {
+    if(this.isLastFrame()) {
+      return null;
+    }
+    if(this.scorecard[this.scorecard.length -2] === [10] || [0, 10]) {
+      let strikeBonus = this.scorecard[this.scorecard.length - 1];
+      this.scorecard[this.scorecard.length - 2] = this.scorecard[this.scorecard.length - 2].concat(strikeBonus);
+    }
+  }
+
+  spareBonus() {
+    if(this.isLastFrame()) {
+      return null;
+    }
+    if(this.scorecard[this.scorecard.length -2] === [10] || [0, 10]) {
```

Comment:
same here. It's also the same condition as the one you used in `strikeBonus()`, is that intentional? It doesn't look like this would describe a spare but you've spent more time with the bowling scoring logic than I! Do your tests currently pass?

If the code in this function is indeed meant to be the same, then you could reduce duplication by extracting the overlapping code into a helper function.

Reply:
I didn't know that === takes precedence yet, tests are passing though so it's working as expected.